### ISSUE TITLE
Crash with 'MinGW64 Compiler (C++)' and New SDK

### ISSUE THREAD

**Issue by Bash885** [jeffsp/matleap#21]
I have been trying  build the mex-file on Windows 10 with the free MinGW64 Compiler (C++)'  and using he latest LeapMotion SDK and keep getting the errors:
"
Error using mex
C:\Users\Bashar\AppData\Local\Temp\mex_2638521891010278_5996\matleap.obj:matleap.cpp:(.text+0x251): undefined reference
to `Leap::Controller::frame(int) const'
C:\Users\Bashar\AppData\Local\Temp\mex_2638521891010278_5996\matleap.obj:matleap.cpp:(.text+0x259): undefined reference
to `Leap::Frame::id() const'
C:\Users\Bashar\AppData\Local\Temp\mex_2638521891010278_5996\matleap.obj:matleap.cpp:(.text+0x27d): undefined reference
to `Leap::Frame::timestamp() const'
.
.
"

I would greatly appreciate any advise on how to fix this issue (is it an SDK compatibility issue, if so which SDK version do you recommend as I tried various SDK versions).

**Comment by jeffsp** [comment]
@Bash885 I don't think that it is an SDK compatibility issue. The linker is not able to find any Leap library functions.

Matleap does not specifically support MinGW64, but I think you can solve your problem by looking at [build.m](https://github.com/jeffsp/matleap/blob/master/build.m).

Lines 13 through 24 set the libdir_switch which needs to point to the Leap library. Later, on line 34, it prints the command that is sent to the mex compiler.

Please examine the output from line 34, `Evaluating "mex  -I./LeapSDK/include -L..."`, and make sure whatever comes after the `-L` points to the correct location of the Leap SDK library.  If it does not, you may need to edit that file and correct it.

Let me know how it goes.

Thanks,
Jeff

**Comment by Bash885** [comment]
Thanks Jeff for the quick response.  

After copying the Matleap file into the desktop and adding the LeapSDK folder, I tried the following in the command line to ensure that the right path to Leap.lib  is included:

mex matleap.cpp -IC:\Users\TwoFlower\Desktop\matleap-master\LeapSDK\include -LC:\Users\TwoFlower\Desktop\matleap-master\LeapSDK\lib\x64 -lLeap

It is still giving the same error (which is by the way the same as if I remove the syntax after "-L...", hence you are right about linker not finding/identifying Leap library). This is very mysterious. Could it be that the compiler is not appropriately reading Leap.lib or the new Leap.lib is not really compatible with Matleap?

Thanks once more for your great help.

**Comment by jeffsp** [comment]
Does Leap support MinGW? I see some very old posts that say that it does not, but I am not sure if more current versions support it. The libraries in LeapSDK\lib\x64 are likely for Linux. Probably for MinGW, you will link with .lib files, which are stubs for their respective .dll... but I'm not 100% certain.

Since you are developing on Windows, you may consider installing the free version of Visual Studio and using that instead of MinGW.

**Comment by joseguerr** [comment]
Sorry to bother you @Bash885 but i have stumbled upon the exact same issue. Did you manage to find a solution for this problem?

Cheers,
José

**Comment by Bash885** [comment]
Dear joseguerr,

No the error persisted and I could not build the mex-file on Windows 10 to be able to use the leap motion data (namely finger positions) in real-time in MATLAB. Let me know please if you find a solution,

Regards,

Bash

**Comment by joseguerr** [comment]
Hey there @Bash885,

I solved the problem by compiling with Visual Studio. In my case i used the 2013 community version and it worked for Matlab 2016a and 2017a. Be careful when running the build.m file because it will sometimes compiling with MinGW64 even if you have Visual Studio as your default compiler (select the Visual Studio compiler again if that's the case). A message will appear just after the final step so you can always check what compiler was used. Good luck!

Cheers,
José Guerra

**Comment by kuldeep429** [comment]
Thanks José Guerra!! it helped!! :)
